src/Features/CSharp/Portable/CodeMapper/CSharpMapCodeService.ReplaceMapperHelper.cs

```diff
@@ -0,0 +1,292 @@
+                    || t.IsKind(SyntaxKind.MultiLineCommentTrivia)
+                    || t.IsKind(SyntaxKind.DocumentationCommentExteriorTrivia)
+                    || t.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia)
+                    || t.IsKind(SyntaxKind.MultiLineDocumentationCommentTrivia)
+                    || t.IsKind(SyntaxKind.SingleLineCommentTrivia);
+        }
+
+        private static bool ShouldSkipAttributes(AttributeSyntax[] insertionAttributes, AttributeSyntax[] targetAttributes)
+        {
+            if (insertionAttributes.Length == 0)
+            {
+                return true;
+            }
+
+            if (insertionAttributes.Length == targetAttributes.Length)
+            {
+                foreach (var insertionAttribute in insertionAttributes)
+                {
+                    // If we find at least one attribute that doesn't exists on target, we'll take it as we need to
+                    // replace those attributes.
+                    if (!targetAttributes.Any(a => a.ToString() == insertionAttribute.ToString()))
+                    {
+                        return false;
+                    }
+                }
+
+                // If all insertion attributes are also located in target attributes, we can skip the attributes.
+                return true;
+            }
+
+            return false;
+        }
+
+        private static AttributeSyntax[] TryGetNodeAttributes(SyntaxNode node)
+        {
+            var insertionAttributes = Array.Empty<AttributeSyntax>();
+            if (node is LocalFunctionStatementSyntax lfs)
+            {
+                insertionAttributes = lfs.AttributeLists.SelectMany(a => a.Attributes).ToArray();
+            }
+            else if (node is MethodDeclarationSyntax mds)
+            {
+                insertionAttributes = mds.AttributeLists.SelectMany(a => a.Attributes).ToArray();
+            }
+            else if (node is ClassDeclarationSyntax cds)
+            {
+                insertionAttributes = cds.AttributeLists.SelectMany(a => a.Attributes).ToArray();
+            }
+
+            return insertionAttributes;
+        }
+
+        private static SyntaxNode? TryRemoveAttributes(SyntaxNode node)
+        {
+            if (node is LocalFunctionStatementSyntax lfs)
+            {
+                return lfs.WithAttributeLists(SyntaxFactory.List<AttributeListSyntax>());
+            }
+            else if (node is MethodDeclarationSyntax mds)
+            {
+                return mds.WithAttributeLists(SyntaxFactory.List<AttributeListSyntax>());
+            }
+            else if (node is ClassDeclarationSyntax cds)
+            {
+                return cds.WithAttributeLists(SyntaxFactory.List<AttributeListSyntax>());
+            }
+
+            return null;
+        }
+
+        private static SyntaxNode? TryAddAttributesAfterLeadingTrivia(SyntaxNode node, IEnumerable<AttributeSyntax> attributes)
```

Comment:
doesn't SyntaxEditor and SyntaxGenerator already do all of this?

Reply:
I wasn't aware of that, if there's a better way of doing it, I think we should be able to explore it! :)
where can I look this up?